src/bin/cql-stress-cassandra-stress/settings/param/multi_param.rs

```diff
@@ -0,0 +1,240 @@
+        required: bool,
+        accept_arbitrary: bool,
+    ) -> Self {
+        Self {
+            prefix,
+            subparams,
+            desc,
+            required,
+            arbitrary_params: if accept_arbitrary {
+                Some(HashMap::new())
+            } else {
+                None
+            },
+            supplied_by_user: false,
+            satisfied: false,
+        }
+    }
+
+    fn accepts_arbitrary(&self) -> bool {
+        self.arbitrary_params.is_some()
+    }
+
+    /// Retrieves arbitrary subparameters (if parsed successfully) and consumes the parameter.
+    pub fn get_arbitrary(self) -> Option<HashMap<String, String>> {
+        if !self.satisfied {
+            return None;
+        }
+        self.arbitrary_params
+    }
+
+    fn try_parse_predefined(&self, arg: &str) -> ParamMatchResult {
+        for param in self.subparams.iter() {
+            let mut borrowed = param.borrow_mut();
+            match borrowed.try_match(arg) {
+                ParamMatchResult::NoMatch => (),
+                ParamMatchResult::Error(e) => return ParamMatchResult::Error(e),
+                ParamMatchResult::Match => match borrowed.parse(arg) {
+                    Ok(()) => return ParamMatchResult::Match,
+                    Err(e) => return ParamMatchResult::Error(e),
+                },
+            }
+        }
+
+        ParamMatchResult::NoMatch
+    }
+
+    fn try_parse_arbitrary(&mut self, arg: &str) -> Result<()> {
+        // Ensure that argument matches pattern "key=value"
+        anyhow::ensure!(
+            ARBITRARY_PARAM.is_match(arg),
+            "Invalid {} specification: {}",
+            self.prefix,
+            arg
+        );
+
+        let split = arg.split('=').collect::<Vec<&str>>();
+        anyhow::ensure!(split.len() == 2, "Invalid arbitrary parameter: {}", arg);
+
+        let map = self.arbitrary_params.as_mut().unwrap();
+        let key = split[0].to_string();
+        let val = split[1].to_string();
+        anyhow::ensure!(
+            !map.contains_key(&key),
+            "{} suboption has been specified more than once",
+            key
+        );
+        map.insert(key, val);
+
+        Ok(())
+    }
+}
+
+impl Param for MultiParam {
+    fn parse(&mut self, arg: &str) -> Result<()> {
+        self.supplied_by_user = true;
+        let arg_val = &arg[self.prefix.len()..];
+
+        // "(some_arg)" -> "some_arg"
+        let arg_val = &arg_val[1..arg_val.len() - 1];
+
+        for (_, [subparam]) in SUBPARAM.captures_iter(arg_val).map(|m| m.extract()) {
+            // Check if the argument matches on of the predefined subparameters.
+            match self.try_parse_predefined(subparam) {
+                ParamMatchResult::Error(e) => return Err(e),
+                ParamMatchResult::Match => continue,
+                _ => (),
+            }
+
+            // If any of the subparamters didn't match, try to parse as arbitrary parameter (if applicable).
```

Comment:
```suggestion
                ParamMatchResult::NoMatch => (),
            }

            // If the subparameter didn't match, try to parse it as arbitrary parameter (if applicable).
```

I think it's clearer written this way - please correct me if I misunderstood the code.

Reply:
I agree